contracts/UBI.sol

```diff
@@ -155,36 +226,34 @@ contract UBI is Initializable {
 
   /** @dev Transfers `_amount` to `_recipient` and withdraws accrued tokens.
   *  @param _recipient The entity receiving the funds.
-  *  @param _amount The amount to tranfer in base units.
+  *  @param _amount The amount to transfer in base units.
   */
   function transfer(address _recipient, uint256 _amount) public returns (bool) {
-    uint256 newSupplyFrom;
-    if (accruedSince[msg.sender] != 0 && proofOfHumanity.isRegistered(msg.sender)) {
-        newSupplyFrom = accruedPerSecond.mul(block.timestamp.sub(accruedSince[msg.sender]));
-        totalSupply = totalSupply.add(newSupplyFrom);
-        accruedSince[msg.sender] = block.timestamp;
+    uint256 _accrued;
+    if(activated[msg.sender] && proofOfHumanity.isRegistered(msg.sender)) {
+      _accrued = (block.timestamp - accruedSince[msg.sender]) * ((accruedPerSecond + stream[address(0)][msg.sender]) - stream[msg.sender][address(0)]);
+    } else if(stream[address(0)][msg.sender] != 0) {
+      _accrued = ((block.timestamp - accruedSince[msg.sender]) * stream[address(0)][msg.sender]);
```

Comment:
It is opinionated to change from `SafeMath` to primitives. Recommend going back to `SafeMath` and using separate PR to make the switch.

Reply:
Safemath costs more gas.. 
I'm still using that for subtraction but not in pre-checked `internal` variables & additions.

We should remove safemath after updating solidity version >=0.8.0 & use `unchecked` for internal "prechecked" variables to save gas.